packages/core/src/connectivity/scp-cf/client-credentials-token-cache.ts

```diff
@@ -38,9 +37,13 @@ const ClientCredentialsTokenCache = (
 
 export function getGrantTokenCacheKey(
   url: string,
-  credentials: ClientCredentials
```

Comment:
Could you add a `TODO` to remove the ClientCredentials in version 2.0 and also remove all internal usages with this API.

Reply:
Done.

Reply:
I added a TODO, but this type is used in other places still and often part of public API. I would not deprecate it yet, because I didn't think the replacement through yet... (e. g. `jwtBearerTokenGrant`).

Reply:
ok please create a follow up ticket containing the points we identified in this review which will be tackeled afterwards. Perhaps by a different colleague to spread the knowledge.